superset-frontend/plugins/plugin-chart-cartodiagram/src/components/OlChartMap.tsx

```diff
@@ -324,54 +340,65 @@ export const OlChartMap = (props: OlChartMapProps) => {
           interaction.setActive(false);
         });
       };
+
       const activateInteractions = () => {
         olMap.getInteractions().forEach(interaction => {
           interaction.setActive(true);
         });
       };
 
+      const { r, g, b, a } = chartBackgroundColor;
+      const cssColor = `rgba(${r}, ${g}, ${b}, ${a})`;
+
       const newChartLayer = new ChartLayer({
         name: CHART_LAYER_NAME,
         chartConfigs: currentChartConfigs,
         chartVizType,
+        chartSizeValues: chartSize.values,
+        chartBackgroundCssColor: cssColor,
+        chartBackgroundBorderRadius,
         onMouseOver: deactivateInteractions,
         onMouseOut: activateInteractions,
+        theme,
       });
 
-      // Only the last setter triggers rerendering of charts
-      newChartLayer.setChartConfig(currentChartConfigs, true);
-      newChartLayer.setChartVizType(chartVizType, true);
-      newChartLayer.setChartSizeValues(chartSize.values);
-
       olMap.addLayer(newChartLayer);
     } else {
+      let recreateCharts = false;
+      if (chartVizType !== chartLayer.chartVizType) {
+        chartLayer.setChartVizType(chartVizType, true);
+        recreateCharts = true;
+      }
+      if (
+        JSON.stringify(currentChartConfigs) !==
```

Comment:
minor: maybe `lodash.isEqual(...)` could be used

https://lodash.com/docs/4.17.15#isEqual